iroh-base/src/key.rs

```diff
@@ -134,14 +134,22 @@ impl PublicKey {
 
     /// Convert to a hex string limited to the first 5 bytes for a friendly string
     /// representation of the key.
-    pub fn fmt_short(&self) -> String {
-        data_encoding::HEXLOWER.encode(&self.as_bytes()[..5])
+    pub fn fmt_short(&self) -> impl Display + 'static {
+        PublicKeyShort(self.0.as_bytes()[0..5].try_into().unwrap())
```

Comment:
Change this to an expect explaining why this is ok.  We tried to remove all unwrap calls at some point (I'm surprised there's no lint).